include/inexor/vulkan-renderer/input/keyboard_mouse_data.hpp

```diff
@@ -13,6 +13,8 @@ class KeyboardMouseInputData {
     std::array<std::int64_t, 2> m_current_cursor_pos{0, 0};
     std::unordered_map<std::int32_t, bool> m_pressed_keys;
     std::unordered_map<std::int32_t, bool> m_pressed_mouse_buttons;
+    bool keyboard_updated = false;
+    bool mouse_buttons_updated = false;
```

Comment:
Not thread safe.

Reply:
Nice you point this out. However the entire `class KeyboardMouseInputData` is not thread safe. We currently have the following warning on it:
```cpp
/// @warning Not thread safe!
```

We will make it thread safe once we begin with parallelization. We need to rethink most code parts then.

Reply:
well for the most parts here we can just use `<atomic>`